iplimit.py

```diff
@@ -0,0 +1,121 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+# by Dan Parsons, aka lerxst, <dparsons@mozilla.com>
+
+from flask import Flask
+from flask import request
+from flaskext.mysql import MySQL
+from netaddr import IPNetwork, IPAddress, core
+import time
+import datetime
+import json
+import ConfigParser
+import optparse
+import sys
+import os
+
+# in days
+DEFAULT_EXCEPTION_LENGTH=2
+
+mysql = MySQL()
+DFORMATTER = "%Y-%m-%d %H:%M:%S"
+TIMEZONE = "GMT %s" % (0 - time.timezone / 60 / 60)
+
+app = Flask(__name__)
+
+IPLIMIT_PROTO="1.0"
+
+mysql.init_app(app)
+
+def load_config(config_file):
+    if not os.path.exists(config_file):
+        print "ERROR: Config file %s not found." % config_file
+        sys.exit(-1)
+    config = ConfigParser.RawConfigParser()
+    config.read(config_file)
+    if 'global' not in config.sections():
+        print "ERROR: [global] section not found in %s." % config_file
+        sys.exit(-1)
+    app.config['MYSQL_DATABASE_USER'] = config.get('global', 'mysql_user')
+    app.config['MYSQL_DATABASE_PASSWORD'] = config.get('global', 'mysql_password')
+    app.config['MYSQL_DATABASE_DB'] = config.get('global', 'mysql_database')
+    app.config['MYSQL_DATABASE_HOST'] = config.get('global', 'mysql_server')
+
+def validIP(ip):
+    try:
+        x = IPAddress(ip)
+    except:
+        return False
+    return True
+
+def process_arguments():
+    parser = optparse.OptionParser(version="%prog 0.1")
+    parser.set_usage("iplimit.py wsgi")
+    parser.add_option('--config', dest='config', default='./iplimit.conf', help='Configuration file (required) (default: ./iplimit.conf)')
+    parser.add_option('--debug', dest='debug', default=False, action="store_true", help="Enable Flask debug mode.")
+    (options, args) = parser.parse_args()
+    return options
+
+@app.route('/')
+def create_exception():
+    # get source IP address. this may need to change depending on how the app is hosted
+    ip = request.remote_addr
```

Comment:
@danparsons When you say it comes via $REMOTE_ADDR is this because this flask app is accessed directly by the user? @gdestuynder had mentioned that this would be behind an Apache reverse proxy (in which case the $REMOTE_ADDR would show as 127.0.0.1)